tests/models/test_taskinstance.py

```diff
@@ -2091,8 +2091,15 @@ def test_execute_queries_count(self, expected_query_count, mark_success):
                 run_type=DagRunType.SCHEDULED,
                 session=session,
             )
+        # an extra query is fired in RenderedTaskInstanceFields.delete_old_records
+        # for other DBs. delete_old_records is called only when mark_success is False
+        expected_query_count_based_on_db = (expected_query_count + 1, expected_query_count)[
+            session.bind.dialect.name in ["postgresql", "sqlite", "mysql"]
+            or expected_query_count == 0
+            or mark_success
```

Comment:
same as above

Reply:
fixed in https://github.com/apache/airflow/pull/9973/commits/6c9df7c658f664c8296af887a2796b3ac082649e